manifests/config.pp

```diff
@@ -93,6 +93,19 @@
     ensure  => absent,
   }
 
+  # adding puma configuration
+  file { "${::foreman::app_root}/config/puma":
+    ensure  => directory,
+  }
+
+  file { "${::foreman::app_root}/config/puma/production.rb":
```

Comment:
I think this should actually live in http://github.com/theforeman/foreman. Since it's Ruby, it can use `ENV` and we can set that in our systemd overrides. This way container deployments can also benefit from this.

Reply:
Maybe like creating `puma/production.rb` in https://github.com/theforeman/foreman/tree/develop/config. If so, where will the template (templates/puma_production.erb) live?

Reply:
I think you can avoid to template it. In pure Ruby:
```ruby
# Configure "min" to be the minimum number of threads to use to answer
# requests and “max” the maximum.
#
# The default is "0, 16".
#
threads ENV.fetch('FOREMAN_PUMA_THREADS_MIN', 0).to_i, ENV.fetch('FOREMAN_PUMA_THREADS_MAX, 16).to_i


# === Cluster mode ===

# How many worker processes to run.
#
# The default is "0".
#
workers ENV.fetch('FOREMAN_PUMA_WORKERS', 4).to_i
```

Untested, but I suspect this will work. Then you should be able to set these environment variables via systemd overrides.

Reply:
@ekohl I tried the configuration you gave above in `/usr/share/foreman/config/puma/production.rb`

I provided overrides in file `/usr/share/foreman-installer/modules/foreman/templates/foreman.service-overrides.erb` like the following:

```
[Service]
User=<%= scope['foreman::user'] %>
Environment=FOREMAN_ENV=<%= scope['foreman::rails_env'] %>
Environment=FOREMAN_HOME=<%= scope['foreman::app_root'] %>
<% if scope['foreman::foreman_service_bind'] -%>
Environment=FOREMAN_BIND=<%= scope['foreman::foreman_service_bind'] %>
<% end -%>
Environment=FOREMAN_PORT=<%= scope['foreman::foreman_service_port'] %>
Environment=FOREMAN_PUMA_WORKERS=8
Environment=FOREMAN_PUMA_THREADS_MIN=4
Environment=FOREMAN_PUMA_THREADS_MAX=12
```

Initially I thought it was working, but later figured that it is not really picking up the environment variable values defined. I am looking into this further.

Reply:
Just adding notes from discussion, I tested this method out and it worked great for me. We will want a Redmine issue since we'll be adding a baseline configuration file to Foreman core codebase as well as this overrides file.